lib/saluki-components/src/sources/checks/execution_context.rs

```diff
@@ -0,0 +1,50 @@
+use std::collections::HashMap;
+
+use saluki_env::{EnvironmentProvider, HostProvider};
+use saluki_metadata;
+use tracing::warn;
+
+// Cache execution information from datadog agent for Python checks
+#[allow(dead_code)] // FIXME temporary
+#[derive(Clone)]
+pub struct ExecutionContext {
+    pub hostname: String,
+    pub http_headers: HashMap<String, String>,
```

Comment:
I would expose getter methods for these so that we don't need to be destructuring `ExecutionContext` every time we want to use it, and updating those destructuring blocks if we want to internally refactor it.

Reply:
I drop this particular change in the rebase, since I misunderstood what you meant there 😅